node-cli/projects.js

```diff
@@ -0,0 +1,107 @@
+const inquirer = require("inquirer");
+const Realm = require("realm");
+const bson = require("bson");
+const index = require("./index");
+const output = require("./output");
+const users = require("./users");
+const main = require("./main");
+const tasks = require("./tasks");
+const manageTeam = require("./manageTeam");
+
+exports.getProjects = async () => {
+  const realm = await index.getRealm(`user=${users.getAuthedUser().id}`);
+  const currentUser = users.getAuthedUser().id;
+  const user = realm.objectForPrimaryKey("User", currentUser);
+  const projects = user.memberOf;
+  output.header("MY PROJECTS:");
+  output.result(JSON.stringify(projects, null, 2));
+};
+
+exports.selectProject = async () => {
+  const realm = await index.getRealm(`user=${users.getAuthedUser().id}`);
+  try {
+    const project = await inquirer.prompt([
+      {
+        type: "input",
+        name: "partition",
+        message: "What is the project partition?",
```

Comment:
Feels a little clunky to make the user copy a full partition key (e.g. `project=5f55572f7240d22e6750ddb5`). Do you think instead we could call `getProjects()` and then present the user a menu of project names that they can move through with arrow keys+enter?

Reply:
Can you help me with the implementation of this? I'm a little stuck.

Reply:
I think this or something similar should work:

```js
exports.selectProject = async () => {
  const realm = await index.getRealm(`user=${users.getAuthedUser().id}`);
  // Copied from getProjects() - we should probably refactor into a reusable function
  const currentUser = users.getAuthedUser().id;
  const user = realm.objectForPrimaryKey("User", currentUser);
  const projects = user.memberOf;
  // Get a list of all the valid project names to show in the menu
  const projectNames = projects.map(p => p.name)
  try {
    // Let the user select a project by name
    const { selectedProjectName } = await inquirer.prompt({
      type: "rawlist",
      name: "selectedProjectName",
      message: "Which project do you want to access?",
      choices: [...projectNames, new inquirer.Separator()],
    });
    // Find the corresponding project document so that we can get the partition value
    const selectedProject = projects.find(p => p.name === selectedProjectName);
    return(projectMenu(selectedProject.partition));
  } catch (err) {
    output.error(JSON.stringify(err));
  }
};
```